Realm/Realm/Linq/RealmResultsVisitor2.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Realms
+{
+    internal class RealmResultsVisitor2 : ExpressionVisitor
+    {
+        private readonly Realm _realm;
+        private readonly RealmObjectBase.Metadata _metadata;
+        private QueryModel _query;
+
+        IQueryableCollection results;
+
+        internal RealmResultsVisitor2(Realm realm, RealmObjectBase.Metadata metadata)
+        {
+            _realm = realm;
+            _metadata = metadata;
+            _query = new QueryModel();
+        }
+
+        protected override Expression VisitMethodCall(MethodCallExpression node)
+        {
+            if (node.Method.DeclaringType == typeof(Queryable))
+            {
+                if (node.Method.Name == nameof(Queryable.Where))
+                {
+                    Visit(node.Arguments[0]);
+                    var whereClause = new WhereClauseVisitor(_metadata);
+                    var lambda = (LambdaExpression)StripQuotes(node.Arguments[1]);
```

Comment:
I would prefer to move this line and the following one to the `WhereClauseVisitor`. It will allow to have the code look very similar for all the clauses. Also, as we use `StripQuotes` in both ClauseVisitors, it would make sense to create a super class for those where we can add these common methods